src/plrust.rs

```diff
@@ -184,18 +197,16 @@ pub(crate) fn compile_function(fn_oid: pg_sys::Oid) -> Result<(PathBuf, String),
             "-Ctarget-cpu=native -Clink-args=-Wl,-undefined,dynamic_lookup",
         )
         .output()
-        .expect("failed to build function shared library");
+        .map_err(PlRustError::CargoBuildExec)?;
 
-    let mut output_string = String::new();
-    unsafe {
-        output_string.push_str(&String::from_utf8_unchecked(cargo_output.stdout));
-        output_string.push_str(&String::from_utf8_unchecked(cargo_output.stderr));
-    }
+    let stdout = String::from_utf8(cargo_output.stdout).map_err(PlRustError::CargoOutputNotUtf8)?;
+    let stderr = String::from_utf8(cargo_output.stderr).map_err(PlRustError::CargoOutputNotUtf8)?;
```

Comment:
I'm pretty sure it will be a cold day in hell before these return anything but, however, this is still better.